### PR TITLE
Clippy lints

### PR DESCRIPTION
**Pull request by allan2**
Hi,

This PR fixes some Clippy lints and also defines lint rules.

- [`map_unwrap_or`](https://rust-lang.github.io/rust-clippy/master/index.html#map_unwrap_or) was not fixed in case maintainers prefer the current style
- [`doc_markdown`](https://rust-lang.github.io/rust-clippy/master/index.html#doc_markdown) was not fixed completely. "MAD_"-prefixed values were not changed  in case maintainers prefer the current style
- [`cast_mut`](https://doc.rust-lang.org/std/primitive.pointer.html#method.cast_mut), used to fix [`ptr_cast_constness`](https://rust-lang.github.io/rust-clippy/master/index.html#/ptr_cast_constness),  bumps the MSRV from 1.60.0 (2022-04-07) to 1.65.0 (2022-11-03).

### CODE REVIEW

**Comment by RazrFalcon** [comment]
Is there a point in marking methods as `const` when `memmap` cannot be used in a const context anyway?

**Comment by allan2** [comment]
Good point. I removed `const fn`.

File: Cargo.toml [hunk 1]
```diff
@@ -18,3 +18,21 @@ libc = "0.2.151"
 [dev-dependencies]
 tempfile = "3"
 owning_ref = "0.4.1"
+
+[lints.clippy]
```
Comment (RazrFalcon) [inline Cargo.toml:22]: This feels like an overkill to me. Do we really need all of them?
I would prefer to keep just the default lints and allow only the one that conflict with the current code. And do so in `lib.rs`.

**Comment by RazrFalcon** [comment]
Also, build is failing on older Rust.

File: Cargo.toml (hunk 1 above)
Comment (allan2) [reply Cargo.toml]: Lint rules have been moved to *lib.rs*. I fixed some more lints. The remaining ones are a bunch of exceptions from the pedantic group.

**Comment by allan2** [comment]
PR updated and ready for review/CI.

File: Cargo.toml [hunk 2]
```diff
@@ -1,7 +1,10 @@
 [package]
 name = "memmap2"
 version = "0.9.5"
-authors = ["Dan Burkert <dan@danburkert.com>", "Yevhenii Reizner <razrfalcon@gmail.com>"]
+authors = [
+	"Dan Burkert <dan@danburkert.com>",
```
Comment (RazrFalcon) [inline Cargo.toml:5]: Please, no tabs...

File: src/lib.rs [hunk 3]
```diff
@@ -1,3 +1,20 @@
+#![deny(clippy::all, clippy::pedantic)]
+#![allow(
```
Comment (RazrFalcon) [inline src/lib.rs:2]: Does any of those checks actually triggered by this crate?

File: clippy.toml [hunk 4]
```diff
@@ -0,0 +1 @@
+msrv = "1.36.0"
```
Comment (RazrFalcon) [inline clippy.toml:1]: New line is missing.

File: src/lib.rs [hunk 5]
```diff
@@ -872,7 +875,7 @@ impl MmapRaw {
     /// but will cause SIGBUS (or equivalent) signal.
     #[inline]
     pub fn as_mut_ptr(&self) -> *mut u8 {
-        self.inner.ptr().cast_mut()
+        self.inner.ptr() as _
```
Comment (RazrFalcon) [inline src/lib.rs:878]: Let's keep the old `as *mut u8` for readability.

File: Cargo.toml (hunk 2 above)
Comment (allan2) [reply Cargo.toml]: Fixed in [d43f291](https://github.com/RazrFalcon/memmap2-rs/pull/125/commits/d43f2910def56b43708e154c480bace88fbc9a70).

File: clippy.toml (hunk 4 above)
Comment (allan2) [reply clippy.toml]: Fixed in [ed732fa](https://github.com/RazrFalcon/memmap2-rs/pull/125/commits/ed732fa27040c8edd520ce3f95a55b9bbf079088).

File: src/lib.rs (hunk 5 above)
Comment (allan2) [reply src/lib.rs]: The [old way](https://github.com/RazrFalcon/memmap2-rs/blob/36ea12019d29f68bec80b6ea8eff2d60a8b571b1/src/lib.rs#L855) was actually `as _`.

Fixed in [d466ca6](https://github.com/RazrFalcon/memmap2-rs/pull/125/commits/d466ca6b8e747b9669639cf95ef3679db6242e88#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R878).

File: src/lib.rs (hunk 3 above)
Comment (allan2) [reply src/lib.rs:2]: All of these exceptions are triggered in the crate. I know it's a long list, but I think the pedantic group is worth it.

**Comment by allan2** [comment]
Thanks for taking the time to review this PR. All requested changes have been implemented.

**Comment by RazrFalcon** [comment]
Thanks!

**Pull request merged**